lambdas/service/app.py

```diff
@@ -745,204 +745,249 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+
+    filter_schema = schema.object(**{
+        facet: schema.optional({
+            'oneOf': [
+                schema.object(**{'is': schema.array({})}),
+                *[
+                    schema.object(**{op: array_pair_spec})
+                    for op in ['contains', 'within', 'intersects']
+                ]
+            ]
+        })
+        for facet in facets
+    })
+    filter_schema['example'] = {'cellCount': {'within': [[10000, 1.e10]]}}
```

Comment:
`1.e10` is a float literal but somehow the example is rendered in the editor as an integer. I'd rather not rely on that conversion (because I don't understand where and why it happens) and instead use an int literal here.